### ISSUE TITLE
GenBank LOCUS line problem from Public JBEI ICE registry

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1656]
Originally raised by @djinnome here:
https://github.com/biopython/biopython/issues/1608#issuecomment-380877275

> Hi folks,
>
>   As promised, [here is an example genbank file](https://github.com/biopython/biopython/files/1904176/JPUB_008754.gb.txt) that generates the following error message when parsed by BioPython
>
> ```python
> from Bio import SeqIO
>
> with open("JPUB_008754.gb", "rU") as input_handle:
>     for record in SeqIO.parse(input_handle, "genbank"):
>         print(record)
> ```
>
> ```
> AssertionError: LOCUS line does not contain space at position 64:
> LOCUS       pEH010                  5743 bp    DNA     circular
> ```
>
> The genbank file was generated from the [Public JBEI ICE registry](https://public-registry.jbei.org/rest/file/8754/sequence/genbank?sid=7d0f4589-2563-4136-ba09-5440d5733bac).
  (you will need to create an account to see the file)
> 
> I have also reported this bug to them so that they can fix it in their genbank exporter.
>
> Sincerely,
>
> Jeremy

**Comment by peterjc** [comment]
Ideally if the LOCUS line is invalid, we generally try to parse what be can that is unambiguous, issue a warning, and continue.

As an aside, the assert statement ought really to be an if statement explicitly raising an exception.

**Comment by kblin** [comment]
Seeing how I poked at the LOCUS line code recently, would you consider the following patch for this?

```patch
diff --git a/Bio/GenBank/Scanner.py b/Bio/GenBank/Scanner.py
index 524f84a15..a273b0881 100644
--- a/Bio/GenBank/Scanner.py
+++ b/Bio/GenBank/Scanner.py
@@ -1277,6 +1277,16 @@ class GenBankScanner(InsdcScanner):
             #    67:68      space
             #    68:79      Date, in the form dd-MMM-yyyy (e.g., 15-MAR-1991)
             #
+            if len(line) < 79:
+                # JBEI genbank files seem to miss a divison code and date
+                # See issue #1656 e.g.
+                # LOCUS       pEH010                  5743 bp    DNA     circular
+                warnings.warn("Truncated LOCUS line found - is this "
+                              "correct?\n:%r" % line, BiopythonParserWarning)
+                padding_len = 79 - len(line)
+                padding = " " * padding_len
+                line += padding
+
             assert line[40:44] in [' bp ', ' aa ', ' rc '], \
                 'LOCUS line does not contain size units at expected position:\n' + line
             assert line[44:47] in ['   ', 'ss-', 'ds-', 'ms-'], \
@@ -1324,7 +1334,8 @@ class GenBankScanner(InsdcScanner):
 
             consumer.molecule_type(line[44:54].strip())
             consumer.topology(line[55:63].strip())
-            consumer.data_file_division(line[64:67])
+            if line[64:76].strip():
+                consumer.data_file_division(line[64:67])
             if line[68:79].strip():
                 consumer.date(line[68:79])
         elif line[self.GENBANK_INDENT:].strip().count(" ") == 0:
diff --git a/Tests/test_GenBank_unittest.py b/Tests/test_GenBank_unittest.py
index 5254dd365..6b96f830c 100644
--- a/Tests/test_GenBank_unittest.py
+++ b/Tests/test_GenBank_unittest.py
@@ -376,6 +376,8 @@ class LineOneTests(unittest.TestCase):
              "circular", "DNA", "BCT"),
             ("LOCUS       SCX3_BUTOC                64 aa            linear   INV 16-OCT-2001",
              "linear", None, "INV"),
+            ("LOCUS       pEH010                  5743 bp    DNA     circular",
+             "circular", "DNA", None),
         ]
         for (line, topo, mol_type, div) in tests:
             scanner = Scanner.GenBankScanner()
```

**Comment by peterjc** [comment]
Something like that looks fine in principle - I've not tried the code though.

**Comment by peterjc** [comment]
Closed via #1713, thanks Kai!
